src/libraries/Common/src/Interop/Windows/Kernel32/Interop.Globalization.cs

```diff
@@ -85,14 +94,14 @@ internal static unsafe partial class Kernel32
                     int cchCount2,
                     bool bIgnoreCase);
 
-        [DllImport("kernel32.dll", EntryPoint = "FindStringOrdinal")]
+        [DllImport("kernel32.dll", EntryPoint = "FindStringOrdinal", SetLastError = SetLastErrorForDebug)]
         internal static extern int FindStringOrdinal(
                     uint dwFindStringOrdinalFlags,
                     char* lpStringSource,
                     int cchSource,
                     char* lpStringValue,
                     int cchValue,
-                    int bIgnoreCase);
+                    bool bIgnoreCase);
```

Comment:
This will add marshaling frame. Does it have effect on performance for small strings?

You can use upper-case `BOOL` to get bool that is both zero-overhead and self-describing.

Reply:
You mean apply `[MarshalAs(BOOL)]` to avoid the marshaling frame?

Reply:
Oh, we have an actual internal `BOOL` enum. Got it.

Reply:
I almost want to define `BOOL.FromBoolean(bool value) => Unsafe.As<bool, BOOL>(ref value)` to avoid the branching at the call sites. But that's probably an issue for a different PR.

Reply:
And it does not work. bool and BOOL have different sizes.

Reply:
Sorry, `Unsafe.As<bool, byte>(ref value)`. The same thing we tried back in https://github.com/dotnet/coreclr/pull/16138 but was killed.